### ISSUE TITLE
Watch hotel config port

### ISSUE THREAD

**Issue by JSn1nj4** [macav/hotelier#3]
Hi. Can Hotelier be made to read `~/.hotel/conf.json` to see if there is a custom port added, and then watch that port instead of the default port 2000?

I'll probably look into this also, but I wanted to at least open the issue for it.

**Comment by macav** [comment]
Hi, thanks for opening the issue! Yes, I had this in mind. Not sure what is the path on Windows, so this would need to be checked as well. 
If you manage to look into it first, i'd be very glad for the PR!

**Comment by JSn1nj4** [comment]
I know in the Windows environment, at least from Command Prompt, the path is like this:
```
%HOMEPATH%\.hotel\conf.json
```
Windows wraps its environment vars in `%`.

**Comment by macav** [comment]
Yes, this is due to the fact that I had problems with CORS and Hotel. There's a reverse proxy set up, so that all the request in development to the same origin go to `localhost:2000` - https://github.com/macav/hotelier/blob/master/package.json#L35

It's not really SSR, but there's electron which is running in the background and react app which is the client. you should have access to `process.env` from anywhere. One thing maybe would help, there's a file which makes some electron API accessible to the client, so that electron & react app can communicate via `ipcRenderer`... this is used for example to open the URL of the hotel server: https://github.com/macav/hotelier/blob/master/public/preload.js
And yes, the tests would be nice :)

Hopefully this helps you a bit.

**Comment by JSn1nj4** [comment]
Ok. Will look at that file then. Thanks!

**Comment by JSn1nj4** [comment]
Just letting you know I haven't given up yet. It's just taking me a while to digest everything.

For the most part, I can tell what's going on in `electron.js`. and I get that I can use the Node API from within `preload.js`. I still need to learn how to use the `ipcMain` and `ipcRenderer` objects, but I think I'm getting there.

**Comment by JSn1nj4** [comment]
@macav may I ask what part this app is considered part of the "renderer" process?

This is where I'm stuck. I keep coming across information saying I should be able to import Node and Electron modules (such as `fs` and `ipcRenderer` ) inside of the renderer process. But trying to import `ipcRenderer` from Electron results in the following error:
```
TypeError: fs.existsSync is not a function
(anonymous function)
<path/to/projects/dir>/hotelier/node_modules/electron/index.js:6
  3 | 
  4 | var pathFile = path.join(__dirname, 'path.txt')
  5 | 
> 6 | if (fs.existsSync(pathFile)) {
  7 |   module.exports = path.join(__dirname, fs.readFileSync(pathFile, 'utf-8'))
  8 | } else {
  9 |   throw new Error('Electron failed to install correctly, please delete node_modules/electron and try installing again')
```

So it's trying to import the Electron module, but it's not able to use the `fs` module, even when I try to import it before importing Electron.

I've tried using both `require` and `import` syntaxes with the same issue.

Current example in [`src/util.js`][1]

Other files tried in:
- [`src/app.js`][2]
- [`src/index.js`][3]

Those last 2 files don't currently have code like what's in `util.js`. That's just where I've tried to import `ipcRenderer` from Electron.

[1]: https://github.com/JSn1nj4/hotelier/blob/feature/read-custom-hotel-port/src/utils.js#L1-L10
[2]: https://github.com/JSn1nj4/hotelier/blob/feature/read-custom-hotel-port/src/app.js
[3]: https://github.com/JSn1nj4/hotelier/blob/feature/read-custom-hotel-port/src/index.js

**Comment by JSn1nj4** [comment]
Ok. So then the React app doesn't have direct access to the renderer. But wouldn't the port value need to be accessible to the React app, for instance, in the [`src/api.js`][1] file?

[1]: https://github.com/JSn1nj4/hotelier/blob/feature/read-custom-hotel-port/src/api.js#L22

**Comment by macav** [comment]
no, directly no. Yes, it would be needed for the API, but you can bind the port to `window` and read it from there. Again, the `shell` example can be useful here. It's bound in `preload.js` and then used in https://github.com/macav/hotelier/blob/master/src/utils.js#L10.

**Comment by JSn1nj4** [comment]
@macav I keep getting `undefined` though when I try to read it from there. Same for `ipcRenderer`.

If I could read the hotel port directly from the window, I wouldn't need `ipcRenderer`, but both of those (when attached to `window` via the preload script) are `undefined` within the React app. I've also confirmed this by `console.log(window)` within the React app and manually inspecting the logged Window object, just in case I was missing something.

From the examples I've looked at, I think this has to do with the React app not having access to the renderer. My understanding is that attaching values to the `window` object that way gives the *renderer* access to those values.

Will have to see if there's another way to handle it (maybe exposing a rest API endpoint for the React app to call, if that's possible with Electron).

**Comment by JSn1nj4** [comment]
I'm sorry to say that I think I need to give this up. I've spent a good deal of time trying to find a workaround, but I think I'm too green to work this deep in an Electron project.

Thanks for the pointers though. I have definitely learned a bit by fiddling with this project.

**Comment by macav** [comment]
@JSn1nj4 Thanks for your help anyway! Could you please open a WIP PR with what you have? Maybe it will be useful and i could continue from that..

**Comment by JSn1nj4** [comment]
@macav sure. I can't do it right this minute, but I'll make sure to do it the first chance I get.

**Comment by macav** [comment]
Solved in v0.2.0, please install it manually :)
